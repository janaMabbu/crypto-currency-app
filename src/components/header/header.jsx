import React from 'react'
import './header.less'

export const Header = () => {
  return (
      <header className="col-sm-12 app-header__content">
        <h1 className="app-header__logo">Crypto Tracker</h1>
      </header>
    )
}