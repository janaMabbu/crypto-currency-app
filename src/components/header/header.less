@import "~less/common.less";

.app-header {
  &__content {
    color: @lightest-grey;
    background: @white-human;
    text-align: left;
    padding: 20px;
  }
}
