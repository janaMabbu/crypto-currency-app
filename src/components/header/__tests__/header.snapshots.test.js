import { shallow } from 'enzyme'
import React from 'react'

import { Header } from 'components/header/header'

describe('Component - Header', () => {
  it('should render the Header snapshot', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()
  })

  it('it should contain logo , banner, and text', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper.find('.app-header__logo').length).toBe(1)
  })

})