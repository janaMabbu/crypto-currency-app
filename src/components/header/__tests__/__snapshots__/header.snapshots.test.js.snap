// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Header should render the Header snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1
        className="app-header__logo"
      >
        Crypto Tracker
      </h1>,
      "className": "col-sm-12 app-header__content",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Crypto Tracker",
        "className": "app-header__logo",
      },
      "ref": null,
      "rendered": "Crypto Tracker",
      "type": "h1",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1
          className="app-header__logo"
        >
          Crypto Tracker
        </h1>,
        "className": "col-sm-12 app-header__content",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Crypto Tracker",
          "className": "app-header__logo",
        },
        "ref": null,
        "rendered": "Crypto Tracker",
        "type": "h1",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
